src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/DoctrineReceiver.php

```diff
@@ -150,4 +146,32 @@ private function createEnvelopeFromData(array $data): Envelope
             new TransportMessageIdStamp($data['id'])
         );
     }
+
+    private function withRetryableExceptionRetry(callable $callable): void
```

Comment:
This code makes me think we need a `symfony/retry` component for handling retry logic. There are a few places in the Symfony codebase that do this sort of thing now. And in my own application code, I have my own class that I use for handling retries. It would be cool to have one component to handle it.

Reply:
Different contexts can make that complicated though maybe? There is https://github.com/symfony/http-client/blob/7.0/Retry/RetryStrategyInterface.php already for example (though of course tailored more specifically for the `HttpClient` component right now).